sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs

```diff
@@ -438,13 +440,32 @@ private static void CloseLink(RequestResponseAmqpLink link)
         /// <param name="outcome"></param>
         /// <param name="timeout"></param>
         private async Task DisposeMessagesAsync(
-            IEnumerable<Guid> lockTokens,
+            Guid[] lockTokens,
             Outcome outcome,
             TimeSpan timeout)
         {
             ThrowIfSessionLockLost();
-            List<ArraySegment<byte>> deliveryTags = ConvertLockTokensToDeliveryTags(lockTokens);
 
+            static List<ArraySegment<byte>> ConvertLockTokensToDeliveryTags(Guid[] tokens)
```

Comment:
I deliberately moved this to a local function. Before it was a static function but since I introduced buffer rental I would have needed to add comments to indicate that users of the method need to return the buffers and since this method was the only user of this code I figured it is much more convenient for this to be a local

Reply:
Actually now that I look at it I think I can even improve this.

Reply:
Pushed a slightly improved version